actions/run-nextstrain-ci-build/action.yaml

```
@@ -0,0 +1,52 @@
+name: run-nextstrain-ci-build
+description: >-
+  Runs a single `nextstrain build` command in a given sub-directory of
+  a pathogen repo. Must be provided with the name of the sub-directory
+  and the runtime to use. Requires that the Nextstrain CLI runtime
+  already be provisioned (e.g., via the `setup-nextstrain-cli` action
+  in this repo).
+
+  Note that this action exists primarily as a means to keep the
+  `pathogen-repo-ci` workflow DRY; it is unlikely to be useful outside
+  the context of that specific workflow.
+
+inputs:
+  directory:
+    description: The name of the sub-directory to run the build from
+    type: string
+    required: true
+  runtime:
+    description: Nextstrain runtime to use for the build
+    type: string
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - id: run-build
+      env:
+        DIR: ${{ inputs.directory }}
+      run: |
+        if [[ -f nextstrain-pathogen.yaml && -f $DIR/Snakefile && -f $DIR/build-configs/ci/config.yaml ]]; then
+          nextstrain check-setup ${{ inputs.runtime }} --set-default
+          nextstrain build $DIR --configfile build-configs/ci/config.yaml
```

Echoing @joverlee521 from [Slack](https://bedfordlab.slack.com/archives/C0K3GS3J8/p1716316838146949?thread_ts=1716315967.004439&cid=C0K3GS3J8): did you consider using a Snakemake target instead of a config file? This would looks something like:

```suggestion
            nextstrain build $DIR ci_target
```

config/defaults.yaml:

```yaml
viruses:
  - "229e"
  - "nl63"
  - "oc43"
  - "hku1"

ci_viruses:
  - "hku1"
```

Snakefile:

```Snakefile
rule ci_target:
    input:
        expand("auspice/{virus}.json", virus=config["ci_viruses"]),
```

---

_Debating with my past self here_

The Snakemake target only works if we are pointing to a target from the default workflow, it would not support using custom rules to copy over example data...

---

See [example in the pathogen-repo-guide](https://github.com/nextstrain/pathogen-repo-guide/blob/80cc1733b47f173772fe45005b82a69c6f8cb6c9/phylogenetic/build-configs/ci/config.yaml)

---

So, to wrap this up: no, I did not consider a Snakemake target; mainly because I was following what I saw in the pathogen-repo-guide. Gonna leave this as-is.